### PR TITLE
Oog create2

### PR DESCRIPTION
**Pull request by lightsing**

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/execution/error_oog_create2.rs [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        TestContext, MOCK_ACCOUNTS,
+    };
+
+    struct TestCase {
+        bytecode: Bytecode,
+        gas: Word,
+    }
+
+    #[test]
+    fn test() {
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/error_oog_create2.rs:268]: have a look at other tests.. May not be a good name

Comment (lightsing) [reply zkevm-circuits/src/evm_circuit/execution/error_oog_create2.rs:268]: tests all cases, what could be a better name?

Comment (lispc) [reply zkevm-circuits/src/evm_circuit/execution/error_oog_create2.rs:268]: test_oog_create2

File: zkevm-circuits/src/evm_circuit/execution/error_oog_create2.rs [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+                + 6.expr() * minimum_word_size.expr(),
+        );
+
+        cb.require_equal(
+            "gas_left must less than gas_cost",
+            insufficient_gas.expr(),
+            true.expr(),
+        );
+
+        // Current call must fail.
```
Comment (silathdiir) [inline zkevm-circuits/src/evm_circuit/execution/error_oog_create2.rs:110]: Should we replace with `CommonErrorGadget`?

**Review by silathdiir — APPROVED** [review]
Except may use `CommonErrorGadget` and rename test to `test_oog_create2`, others LGTM 👍

**Review by silathdiir — APPROVED** [review]
LGTM 👍

**Pull request merged**